gix-ref/src/store/file/overlay_iter.rs

```diff
@@ -195,10 +195,9 @@ impl Platform<'_> {
         self.store.iter_packed(self.packed.as_ref().map(|b| &***b))
     }
 
-    /// As [`iter(…)`][file::Store::iter()], but filters by `prefix`, i.e. "refs/heads".
-    ///
-    /// Please note that "refs/heads" or "refs\\heads" is equivalent to "refs/heads/"
-    pub fn prefixed(&self, prefix: &Path) -> std::io::Result<LooseThenPacked<'_, '_>> {
+    /// As [`iter(…)`][file::Store::iter()], but filters by `prefix`, i.e. "refs/heads/" or
+    /// "refs/heads/feature-".
+    pub fn prefixed(&self, prefix: Cow<'_, BStr>) -> std::io::Result<LooseThenPacked<'_, '_>> {
```

Comment:
I think it's fair to have this breaking change, as this prefix is essentially a `RepositoryPath`, i.e. a slash-separated relative path.
Something I think that should come with it is tests that show that behaviour.

Reply:
The tests are now more explicitly named to test that.